File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/portlet/PortletFileUpload.java
package org.apache.commons.fileupload.portlet;

import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.FileItemFactory;
import org.apache.commons.fileupload.FileUpload;
import org.apache.commons.fileupload.IntegrationUtils;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public class PortletFileUpload extends FileUpload {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/portlet/PortletFileUpload.py",
              "PortletFileUpload"));

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  public static PortletFileUpload PortletFileUpload1() {

    return new PortletFileUpload(null);
  }

  public PortletFileUpload(FileItemFactory fileItemFactory) {

    super(0, fileItemFactory);
  }

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {
    super.pyToJ(idMap);

    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {
    super.jToPy(idMap);
    this.obj.invokeMember("__setattr__", "javaObj", this);

    return idMap;
  }
}

File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/portlet/PortletFileUpload.py
from __future__ import annotations
import io
from src.main.org.apache.commons.fileupload.FileItemFactory import *
from src.main.org.apache.commons.fileupload.FileUpload import *
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class PortletFileUpload(FileUpload, metaclass=StaticFieldRedirector):
    @staticmethod

    def PortletFileUpload1() -> PortletFileUpload:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(PortletFileUpload.javaClz.PortletFileUpload1(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass

            
    def __init__(self, fileItemFactory: FileItemFactory) -> None:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        translatedArg0 = JavaHandler.valueToObject(fileItemFactory, "FileItemFactory", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj = PortletFileUpload.javaClz(translatedArg0)
            self.javaObj.setPythonObject(self)
            self.javaObj.jToPy(idMapJToPy)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            JavaHandler.mapping(translatedArg0, idMapJToPy, fileItemFactory)
    javaClz = java.type("org.apache.commons.fileupload.portlet.PortletFileUpload")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))